pkg/v1/providers/yttcb/clusterbootstrap.yaml

```diff
@@ -0,0 +1,128 @@
+#@ load("@ytt:data", "data")
+#@ load("lib/helpers.star", "get_tkr_version_from_tkr_name")
+
+#@ def antrea_configs_exist():
+#@   return (data.values.NSXT_POD_ROUTING_ENABLED or
+#@           data.values.ANTREA_NO_SNAT or
+#@           data.values.ANTREA_TRAFFIC_ENCAP_MODE != "encap" or
+#@           not data.values.ANTREA_PROXY or
+#@           not data.values.ANTREA_ENDPOINTSLICE or
+#@           not data.values.ANTREA_POLICY or
+#@           data.values.ANTREA_NODEPORTLOCAL or
+#@           not data.values.ANTREA_TRACEFLOW or
+#@           data.values.ANTREA_EGRESS or
+#@           data.values.ANTREA_FLOWEXPORTER or
+#@           data.values.ANTREA_DISABLE_UDP_TUNNEL_OFFLOAD)
+#@ end
+
+#@ def vspherecsi_configs_exist():
+#@   return (not data.values.VSPHERE_INSECURE or
+#@           data.values.USE_TOPOLOGY_CATEGORIES)
+#@ end
+
+#@ def should_create_clusterbootstrap():
+#@   return data.values.CNI != "antrea" or antrea_configs_exist() or vspherecsi_configs_exist()
+#@ end
+
+#@ if should_create_clusterbootstrap():
+---
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+  name: #@ data.values.CLUSTER_NAME
+  namespace: #@ data.values.NAMESPACE
+  annotations:
+    tkg.tanzu.vmware.com/add-missing-fields-from-tkr: #@ "{}".format(get_tkr_version_from_tkr_name(data.values.KUBERNETES_RELEASE))
+spec:
+  cni:
+    #@ if data.values.CNI == "antrea":
+    refName: antrea*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: AntreaConfig
+        name: #@ data.values.CLUSTER_NAME
+    #@ elif data.values.CNI == "calico":
+    refName: calico*
+    valuesFrom:
+      providerRef:
+        apiGroup: cni.tanzu.vmware.com
+        kind: CalicoConfig
+        name: #@ data.values.CLUSTER_NAME
```

Comment:
nit: you can probably put the elseif on  line 44 and lleave "name: data.values.CLUSTER_NAME" for both (single line).